src/trace/layers/hashed.rs

```diff
@@ -398,17 +403,17 @@ impl<K: HashOrdered+Clone+Default, L: MergeBuilder> HashedBuilder<K, L> {
 
 /// A cursor with a child cursor that is updated as we move.
 #[derive(Debug)]
-pub struct HashedCursor<K: HashOrdered, L: Cursor> {
+pub struct HashedCursor<B, K: HashOrdered, L: Cursor> {
 	shift: usize,			// amount by which to shift hashes.
 	bounds: (usize, usize),	// bounds of slice of self.keys.
 	pos: usize,				// <-- current cursor position.
 
-	keys: Rc<Vec<Entry<K>>>,
+	keys: OwningRef<Rc<B>, Vec<Entry<K>>>,
```

Comment:
This seems like it could possibly be `keys: B` with the constraint `B: Deref<Target=[Entry<K>]>`, which seems appealing in that it hides the details of how we've chosen to wrap up the owned reference to `[Entry<K>]`.